### PR TITLE
patch: Migrate balenaEngine Debugging docs from masterclass

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Hello folks, I bring docs.
These docs were part of the debugging masterclass.
I am trying to make debugging masterclasses render dynamically from the content present in main docs as much as they can.
This lead to me transferring docs to their rightful places.
Plan is to have these docs live in the balenaEngine repo, and then main docs pulling these in to generate the masterclass.

What I need help in:

1. Product owners giving these docs a once over to update outdated information or
2. Straight up, merge this PR.

I am transferring these docs straight from the debugging masterclass so please leave the headings how they currently are. Anything else can be modified.
Please commit directly to make the changes on this PR. I don't have the bandwidth to resolve issues with the docs themselves.

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipul@balena.io>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
Another PR @lmbarros merge it whenever you get a chance.

**Comment by vipulgupta2048** [comment]
Reping @lmbarros on merging this or make changes as needed

File: balena-docs/engine-debugging.md [hunk 1]
```diff
@@ -0,0 +1,487 @@
+### 9. Working with balenaEngine
+
+Service: `balena.service`
+
+balenaEngine is balena's fork of Docker, offering a range of added features
+including real delta downloads, minimal disk writes (for improved media wear)
```
Comment (lmbarros) [inline balena-docs/engine-debugging.md:6]:
```suggestion
including real delta downloads, minimal disk writes (for reduced media wear)
```

File: balena-docs/engine-debugging.md [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+]
+```
+
+Of particular interest here is the `"RootFS"` section, which shows all of the
+layers that make up an image. Without going into detail (there are plenty of
+easily Google-able pages describing this), each balena image is made up of a
+series of layers, usually associated with a `COPY`, `ADD`, `RUN`, etc.
+Dockerfile command at build time. Each layer makes up part of the images
+filing system, and when a service container is created from an image, it uses
+these layers 'merged' together for the underlying filing system.
```
Comment (lmbarros) [inline balena-docs/engine-debugging.md:271]:
```suggestion
Of particular interest here is the `"RootFS"` section, which shows all of the
layers that make up an image. Without going into detail (there are plenty of
easily Google-able pages describing this), each balena image is made up of a
series of layers, usually associated with a `COPY`, `ADD`, `RUN`, etc.
Dockerfile command at build time. Each layer makes up part of the images
file system, and when a service container is created from an image, it uses
these layers 'merged' together for the underlying file system.
```

File: balena-docs/engine-debugging.md [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+with the `diff` file result, as this describes the diff for the layer:
+
+```shell
+root@debug-device:/var/lib/docker/image/overlay2/layerdb/sha256# cat 09b78bc523987759f021e0a0e83e69c8084b1e3c20f14b4bb9534f3cdcc6ac3c/cache-id
+f77ebf3b4836d289c2515c82537cd774354b7342c2a4899fcffb51ac23e9e9b7
+```
+
+We now have the corresponding `cache-id` for the layer's directory layout,
+and we can now examine the file system for this layer (all the diffs are
+store in the `/var/lib/docker/overlay2/<ID>/diff` directory):
```
Comment (lmbarros) [inline balena-docs/engine-debugging.md:350]:
```suggestion
and we can now examine the file system for this layer (all the diffs are
stored in the `/var/lib/docker/overlay2/<ID>/diff` directory):
```

**Comment by lmbarros** [comment]
LGTM

**Pull request merged**